thirdeye/thirdeye-frontend/app/pods/components/rootcause-select-comparison-range/component.js

```diff
@@ -36,39 +54,154 @@ export default Component.extend({
+    return this.get('compareMode');
+  }),
+
+  /**
+   * Default after elements are inserted
+   */
+  didInsertElement() {
+    this._super(...arguments);
+    let $range = $('.js-range-slider');
+    const timeFormat = this.get('timeFormat');
+    const { compareMode, onChange } = this.getProperties('compareMode', 'onChange');
+
+    $range.ionRangeSlider({
+      type: 'double',
+      grid: true,
+      grid_num: 1,
+      hide_min_max: true,
+      step: this.get('granularityOneWay'),
+      min: this.get('minDisplayWindow').format('X'),
+      max: this.get('maxDisplayWindow').format('X'),
+      from: this.get('minInvestigatePeriod').format('X'),
+      to: this.get('maxInvestigatePeriod').format('X'),
+      prettify: function (num) {
+        return moment(num, 'X').format(timeFormat);
+      },
+      onFinish: function (data) {
+        // Update the display window's investigation period on the chart
+        onChange(moment.unix(data.from).valueOf(), moment.unix(data.to).valueOf(), compareMode);
+      }
+    });
+
+    // Save slider instance to var
+    this.set('slider', $range.data('ionRangeSlider'));
+  },
+
+  didRender() {
+    this._super(...arguments);
+
+    // Set oneway assignment from these existing CPs
+    this.setProperties({
+      startFormattedOneWay: this.get('startFormatted'),
+      endFormattedOneWay: this.get('endFormatted'),
+      maxDateFormattedOneWay: this.get('maxDateFormatted'),
+      // granularityOneWay: this.get('granularity')
+    });
+
+    // Save original investigation periods
+    this.setProperties({
+      originalMinInvestigatePeriod: this.get('minInvestigatePeriod'),
+      originalMaxInvestigatePeriod: this.get('maxInvestigatePeriod')
+    });
+
+    // Update the slider, by calling it's update method
+    this.get('slider').update({
+      step: this.get('granularityOneWay')
+    });
+  },
+
   actions: {
     onRange(start, end) {
       const { compareMode, onChange } = this.getProperties('compareMode', 'onChange');
+
+      // Update anomalyRange for computed to recalculate
+      this.set('anomalyRange', [moment(start).valueOf(), moment(end).valueOf()]);
+
+      // Investigation period changed on date picker. Update the slider's min and max.
+      this.get('slider').update({
+        from: this.get('minInvestigatePeriod').format('X'),
+        to: this.get('maxInvestigatePeriod').format('X')
+      });
+
+      // Update the display window's investiation period on the chart
+      onChange(moment(start).valueOf(), moment(end).valueOf(), compareMode);
+    },
+
+    onPickerRange(type, time) {
+      const { compareMode, onChange } = this.getProperties('compareMode', 'onChange');
+
+      // Update anomalyRange for computed to recalculate
+      const start = type === 'start' ? moment(time).valueOf() : this.get('minInvestigatePeriod').valueOf();
+      const end = type === 'start' ? this.get('maxInvestigatePeriod').valueOf() : moment(time).valueOf();
```

Comment:
should this be `type === 'end'`?

Reply:
No. If it's `type === 'start'` we want to do the truthty part of the uni operator for both consts.